### PR TITLE
Fixed ~/.pavilion and log issues.

### LINKED ISSUE #87: .pavilion directory creation
When initially running any pav command, ensure it creates the `~/.pavilion` directory.  If no such directory is found and creation doesn't work (or something is stopping it), ensure that the error command is helpful.

### PR DESCRIPTION
**Pull request by Paul-Ferrell**
- ~/.pavilion is created automatically if it doesn't exit.
 - If we can't find a user's home dir, use /tmp/$USER/.pavilion
 - Added warning messages if we can't create various log files.

Fixes #87 and #91

### CODE REVIEW

File: lib/pavilion/config.py [hunk 1]
```diff
@@ -17,11 +17,13 @@
 # Figure out what directories we'll search for configuration files.
 PAV_CONFIG_SEARCH_DIRS = [Path('./').resolve()]
 
-if 'HOME' in os.environ:
-    USER_HOME_PAV = Path(os.environ['HOME'], '.pavilion')
-    PAV_CONFIG_SEARCH_DIRS.append(USER_HOME_PAV)
-else:
-    USER_HOME_PAV = None
+try:
+    USER_HOME_PAV = Path('~/.pavilion').expanduser()
```
Comment (francinelapid) [inline lib/pavilion/config.py]: `AttributeError: 'PosixPath' object has no attribute 'expanduser'`

I think you mean `.home()`?

File: bin/pav.py [hunk 2]
```diff
@@ -93,55 +96,63 @@ def record_factory(*fargs, **kwargs):
...
-        maxBytes=20 * 1024 **2,
-        backupCount=3,
-    )
-    exc_handler.setFormatter(logging.Formatter(
-        "{asctime} {message}",
-        style='{',
-    ))
-    exc_logger.setLevel(logging.ERROR)
-    exc_logger.addHandler(exc_handler)
+        pav_cfg.exception_log.touch()
```
Comment (francinelapid) [inline bin/pav.py]: Still getting this error when running the Travis tests:

```
- Could not create exception log at '~/.pavilion/working_dir/exceptions.log': [Errno 2] No such
- file or directory: '~/.pavilion/working_dir/exceptions.log'
```

Comment (Paul-Ferrell) [reply bin/pav.py]: oh, hah.

File: bin/pav.py [hunk 3]
```diff
@@ -93,55 +106,74 @@ def record_factory(*fargs, **kwargs):
...
-                                       backupCount=3)
-    file_handler.setFormatter(logging.Formatter(pav_cfg.log_format,
-                                                style='{'))
-    file_handler.setLevel(getattr(logging,
-                                  pav_cfg.log_level.upper()))
-    root_logger.addHandler(file_handler)
+    try:
+        log_fn.touch()
+    except (PermissionError, FileNotFoundError) as err:
+        output.fprint("Could write create pavilion log at '{}': {}"
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply bin/pav.py]: I've fixed this.

File: lib/pavilion/config.py (hunk 1 above)
Comment (Paul-Ferrell) [reply lib/pavilion/config.py]: Expand user isn't in 3.4, apparently.

**Comment by Paul-Ferrell** [comment]
I think everything should be goo to go for real now.

**Pull request merged**